backend/authBackend.js

```diff
@@ -1,65 +1,30 @@
-// JWT is our primary identification mechanism
-import * as jwt from 'jsonwebtoken';
+import { getCurrentUser } from '@reshuffle/server-function';
+const { REACT_APP_VALID_HOSTED_DOMAIN } = process.env;
 
-// used to validate incoming Google profiles from client
-import { OAuth2Client } from 'google-auth-library';
+// validates if email is the right REACT_APP_VALID_HOSTED_DOMAIN
+export async function validateUser() {
+  const profile = getCurrentUser(true);
+  let email = profile.emails[0].value;
+  email = email.split('@');
 
-// we depend on both OAUTH_CLIENT_ID and a SECURE
-// HMAC which is used to sign our tokens
-//
-// THE HMAC CANNOT BE LEAKED!!!!!!!!!!!!!!
-const {
-  REACT_APP_OAUTH_CLIENT_ID,
-  JWT_HMAC_KEY,
-  REACT_APP_VALID_HOSTED_DOMAIN,
-} = process.env;
-
-const client = new OAuth2Client(REACT_APP_OAUTH_CLIENT_ID);
+  if (email[1] === REACT_APP_VALID_HOSTED_DOMAIN) {
+    return true;
+  }
 
-/**
- * Makes the necessary requests to Google auth servers
- * to verify that the client user is actually who they
- * say they are.
- *
- * @param { string } token - idToken returned by clientside auth
- */
-async function verifyGoogleUser(token) {
-  const ticket = await client.verifyIdToken({
-    idToken: token,
-    audience: REACT_APP_OAUTH_CLIENT_ID,
-  });
-  return ticket.getPayload();
+  throw new Error(
+    `User cannot be authenticated, ${email[1]} is not a valid host domain`,
+  );
 }
 
-/**
- * Authenticates a Google user whose token was returned
- * from the client-side Google authentication flow. Assuming
- * the user is valid, a side-wide JWT will be returned for
- * future identification.
- *
- * @param { string } googleToken - googleToken returned by clientside auth
- *
- * @return { string } - site-wide JWT that allows future access
- */
-/* @expose */
-export async function authenticateUser(googleToken) {
-  const payload = await verifyGoogleUser(googleToken);
-  const { hd } = payload;
-  if (hd === REACT_APP_VALID_HOSTED_DOMAIN) {
-    // store some basic (unreliable) profile information in the token
-    return jwt.sign({
-      name: payload.name,
-      email: payload.email,
-    }, JWT_HMAC_KEY);
-  } else {
-    // TODO: Format error to meet specification defined in contentBackend
-    throw new Error(`User cannot be authenticated, ${hd} is not a valid host domain`);
+// validates if email is the right REACT_APP_VALID_HOSTED_DOMAIN
+export async function validate() {
```

Comment:
I'm not even really sure both of these functions need to exist. They seem a bit redundant now.